components/logins/src/logins.udl

```diff
@@ -2,18 +2,26 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-namespace logins {
+namespace logins { };
 
-    [Throws=LoginsStorageError]
-    string open_and_get_salt(string path, [ByRef] string encryption_key);
-
-    [Throws=LoginsStorageError]
-    void open_and_migrate_to_plaintext_header(string path, [ByRef] string encryption_key, [ByRef] string salt);
+// Converted from mozilla.appservices.logins.protobuf.rs
+dictionary Login {
+    string id;
+    string origin;
+    string password_enc;
+    string username_enc;
+    string? http_realm;
+    string? form_action_origin;
+    string username_field;
+    string password_field;
+    i64 times_used;
+    i64 time_created;
+    i64 time_last_used;
+    i64 time_password_changed;
 };
 
-
 // Converted from mozilla.appservices.logins.protobuf.rs
-dictionary Login {
+dictionary LoginPayload {
```

Comment:
I like the idea of `Login` and `LoginPayload` given that we can have the payload be a reduced set of params that is needed to be sent over by the consumer, however I see that `LoginPayload` still contains all the fields -- will this be going away? I would imagine it would like something like:

```
LoginPayload {
    string id;
    string hostname;
    string password;
    string username;
    string? http_realm;
    string? form_submit_url;
    string username_field;
    string password_field;
}
```

but I'm probably missing something here, mainly my initial thoughts from my current mental map!

Reply:
If you look in the login.rs file of the per-field-encryption branch, [the Login struct](https://github.com/mozilla/application-services/blob/18e48dd16b07272a6519a4fe0844d3d3a98eb918/components/logins/src/login.rs#L241) has the encrypted username and password fields and the `LoginPayload` struct is for the unencrypted fields. I do like separating those out so the consumers understand which data they're dealing with. But also, I think that the `username_enc` and `password_enc` fields will end up being optional strings, or diverging from `Logins` in some other way (particularly when the scrubbing logic is added). Having said that, it may be that using one struct is fine for now or that `LoginPayload` isn't the correct nomenclature. Either way, open to suggestions

Reply:
`LoginPayload` was designed to be the data we got from the server, which was why the username/password were plaintext.  I think we need to keep that struct as-is (or maybe someone who knows serde more can implement that code in a batter way).

I do like idea of a `Login`-like struct with reduced fields that's intended for the consumer code.  It would just need to be separate from `LoginPayload`

Reply:
I'm not sure which fields are expected and which we can hide, I know that `time_password_changed` needs to be exposed for iOS and I think one of the other timestamp fields might be used for ordering.

Reply:
I made some updates to the exposed login structs. I removed `LoginPayload`, added a `LoginFields` struct with the meta fields removed for the `add_or_update` function, and then added a `DecryptedLogin` struct for the `decrypt_and_fixup_login` function. The meta fields are still needed for the fennec to fenix migration code, but we can consider removing those when we stop supporting that logic.